src/plugins/json-schema-validator/oas3-schema.yaml

```diff
@@ -37,9 +39,9 @@ definitions:
   Reference:
     type: object
     required:
-    - $ref
-    properties:
-      $ref:
+    - "$ref"
+    patternProperties:
+      "^\\$ref$":
```

Comment:
should this be `Reference.properties.$ref` instead of `Reference.patternProperties.^\\$ref$`?

Reply:
Yes. 

Some JSON Schema draft-04 engines get tripped up on `$ref` as a property name, because draft-04 incorporated JSON Reference without adding any guards as to _where_ $ref is semantically valid as a Reference vs it's a property name for an object.

AJV appears to handle this without any issues (while it _did_ have issues with the regex workaround), so we're going to use `Reference.properties.$ref` here.

CC @webron!